[quant] qtensor resize

z-a-f opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #36450 [quant] quantized reflection_pad1d
* #36449 [quant] Generalizing _calculate_dynamic_qparams in quantized test
* **#36442 [quant] qtensor resize**

Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

Commit:
[quant] qtensor resize

[ghstack-poisoned]

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

jerryzh168 commented:
Can you check if we can add this to graph mode? is this also compiled to `CallFunction`?

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

z-a-f commented:
> Can you check if we can add this to graph mode? is this also compiled to `CallFunction`?

It doesn't have the CallFunction. Also, added an entry to quantization.cpp

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/passes/quantization.cpp:
```diff
@@ -118,6 +118,7 @@ std::vector<std::string> _single_input_general_aten_funcs = {
     "upsample_bicubic2d",
     "dropout",
     "reshape",
+    "resize_",
```
what about "resize"?

z-a-f replied:
`resize` is a python module, and script does not see it:

```python
class Model(torch.nn.Module):
  def __init__(self, sz):
    super(Model, self).__init__()
    self.sz = sz
  def forward(self, x):
    return x.resize(self.sz)
torch.jit.script(Model(42))
```

will throw an error:

```console
RuntimeError: 
Tried to access nonexistent attribute or method 'resize' of type 'Tensor'.:
  File "<ipython-input-11-85bfdcc7b088>", line 6
  def forward(self, x):
    return x.resize(self.sz)
           ~~~~~~~~ <--- HERE
```

jerryzh168 replied:
I see, looks like non-inplace resize is deprecated: https://github.com/pytorch/pytorch/blob/master/torch/tensor.py#L351

could you add a comment here for it?

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]

Commit:
Update on "[quant] qtensor resize"


Differential Revision: [D20984080](https://our.internmc.facebook.com/intern/diff/D20984080)

[ghstack-poisoned]
